### ISSUE TITLE
Provide an option to show the entire contents of the linked documents in Backlinks

### ISSUE THREAD

**Issue by ecvgit** [org-roam/org-roam#245]
### Brief Abstract
I create tags in my org mode documents. However, when I navigate to the note for a tag, I only see a small portion of the original document in the back links. It would be good to provide an option to show the entire content of the document in the backlinks. So that one can see the entire document that is being linked to without going back and forth.
### Long Description
Let us say I have a document like 
#+Title: Title 1
tags :: tag1, tag2
** Heading 2
  Some content 3
** Heading 3
 ** Some content 3

Now, when I navigate to tag1, the backlinks section in org-roam buffer shows only the title and tags line. I want it to show the entire contents of Title 1 document when necessary.

### Proposed Implementation (if any)
Not sure how to implement this. May be something like org-show-context-detail can fix this?

### Please check the following:
- [x] No similar feature requests

**Comment by jethrokuan** [comment]
This will likely destroy the formatting of the org-roam buffer. Rather than view the full contents within the narrow sidebar, wouldn't just opening the file be better?

**Comment by ecvgit** [comment]
My use case is this:
Let us say I have a tag called 'China' - I want to see all the notes I have written on china without having to navigate back and forth between individual notes. Just having the title does not help. 
I think this will be a powerful feature.

Or one can provide a separate command, that will collect all the backlinks in a new buffer with the entire contents, which the user can export, save, etc. 

I understand your point about it destroying the formatting. But may be there is a way to work around this by providing making it a sub node of a new node (or something like this)?

**Comment by jethrokuan** [comment]
Yeah okay, got it. It'll probably be a separate interface (search interface, that will allow complex queries like has tag "china" and tag "korea").

**Comment by ecvgit** [comment]
Thank you. That was precisely what I was thinking about. 

Also, I think there is value to having an option similar to the current setup as well (where only related context is shown). So it would be nice to make it an option.

**Comment by herbertjones** [comment]
Here's a proof of concept that provides `org-roam-search`.  It allows searching for all backlinks to a file in a new buffer, and has a toggle to insert just the reference line, or the entire article.

https://gist.github.com/herbertjones/a55131b7bb9e74d61159f5ba14613ef8

**Comment by jethrokuan** [comment]
@herbertjones nice! This was how I imagined it to be as well (inspired from https://github.com/Wilfred/deadgrep)

**Comment by Wetlize** [comment]
@herbertjones @jethrokuan This UI is pretty good. It would be also great to have "heading" (or "outline") restriction option. In such case it would only pull-in trees (instead the whole file content) the heading of which contains the matched backlink . This is useful for journals and the workflow more concretely described [here](https://medium.com/my-learning-journal/how-i-use-roam-as-a-digital-bullet-journal-df6e51e56f0f).

So if you have the next structure in your "journal" file:
```
* [[file::thoughts.org][Thoughts]]
... content ...

* [[file::learned.org][What I've learned]]
... content...
```
that you create daily (or occasionally) using `org-roam-today`, I only would like to pull-in headings (and their content) that match the "Thoughts" backlink.

**Comment by jethrokuan** [comment]
This is next up on the list (revamping the org-roam-buffer)

**Comment by progfolio** [comment]
I have an idea for a flexible implementation of org-roam-buffer. I'll work up a rough sketch.

**Comment by NickSeagull** [comment]
Any progress on this? I'm interested in this feature, and I would like to help :)

**Comment by ahmed-shariff** [comment]
I use the following to view the results of a whole subtree
```emacs-lisp
(defun org-roam-subtree-aware-preview-function ()
    "Same as `org-roam-preview-default-function', but gets entire subtree in specific buffers."
    (if (--> (org-roam-node-at-point)
             (org-roam-node-file it)
             (or (member it
                         ;; This is a list of buffers where I want to see preview of subtree
                         org-roam-subtree-aware-preview-buffers) 
                 (f-ancestor-of-p bibtex-completion-notes-path it)))
        (let ((beg (progn (org-roam-end-of-meta-data t)
                          (point)))
              (end (progn (org-previous-visible-heading 1)
                          (org-end-of-subtree)
                          (point))))
          (-reduce 
           (lambda (str el)
             ;; remove properties not interested. If prop drawer is empty at the end, remove drawer itself
             (s-replace-regexp (format "\n *:%s:.*$" el) "" str))
           ;; remove links
           (list (s-replace-regexp "\\[id:\\([a-z]\\|[0-9]\\)\\{8\\}-\\([a-z]\\|[0-9]\\)\\{4\\}-\\([a-z]\\|[0-9]\\)\\{4\\}-\\([a-z]\\|[0-9]\\)\\{4\\}-\\([a-z]\\|[0-9]\\)\\{12\\}\\]"
                                   ""
                                   (string-trim (buffer-substring-no-properties beg end)))
                 "INTERLEAVE_PAGE_NOTE" "BRAIN_CHILDREN" okm-parent-property-name "PROPERTIES:\n *:END")))
      (org-roam-preview-default-function)))

  (setq org-roam-preview-function #'org-roam-subtree-aware-preview-function)
```

**Comment by hpfr** [comment]
I think this is still relevant. Curious if org-roam-ql enables this; it seems like a flexible way to implement custom views.
